lib/TPP/Transforms/CombineXsmmPass.cpp

```diff
@@ -0,0 +1,179 @@
+  return rewriter.getArrayAttr(gemmFlag);
+}
+
+struct CombineXsmmOp : public OpRewritePattern<xsmm::BrgemmOp> {
+  using OpRewritePattern<xsmm::BrgemmOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(xsmm::BrgemmOp brgemmOp,
+                                PatternRewriter &rewriter) const override {
+    auto *output = brgemmOp.getOperand(3).getDefiningOp();
+    if (!output)
+      return failure();
+
+    // First, match the required fused ops
+    auto result = xsmm::utils::getFusedBrgemmSequenceFromProducer(output);
+    if (failed(result))
+      return failure();
+    auto fusedMatch = *result;
+    // TODO: Support BRGEMM + BINARY && BRGEMM + UNARY patterns
+    if (!fusedMatch.binaryOp || !fusedMatch.unaryOp)
+      return failure();
+    // Validate broadcast flags
+    auto unaryFlags =
+        xsmm::utils::getUnaryFlags(fusedMatch.unaryOp.getOperand(0).getType(),
+                                   fusedMatch.unaryOp.getOperand(2).getType());
+    if (unaryFlags != mlir::xsmm::UnaryFlags::BCAST_SCALAR &&
+        unaryFlags != mlir::xsmm::UnaryFlags::NONE)
+      return failure();
+
+    // TODO: Support more than just COL_0 BCAST
+    auto binaryFlags =
+        xsmm::utils::getBinaryFlags(fusedMatch.binaryOp.getOperand(1).getType(),
+                                    fusedMatch.binaryOp.getOperand(3).getType(),
+                                    mlir::xsmm::utils::OperandPos::LHS);
+    if (binaryFlags != mlir::xsmm::BinaryFlags::BCAST_COL_IN_0)
+      return failure();
+
+    // Now, replace the ops with a fused BRGEMM
+    auto dtype =
+        xsmm::utils::getDataType(rewriter, brgemmOp.getOperand(1).getType());
+    IntegerType integer64 = IntegerType::get(rewriter.getContext(), 64);
+
+    Location loc = brgemmOp.getLoc();
+    auto dims = getSizesAndLeadingDimForBrgemmOp(rewriter, brgemmOp);
+    auto memrefB = brgemmOp.getOperand(2);
+    int64_t batchSize = memrefB.getType().cast<ShapedType>().getShape()[0];
+
+    OpBuilder::InsertionGuard guard(rewriter);
+    rewriter.setInsertionPointAfter(fusedMatch.binaryOp);
+    Value dispatched = rewriter.create<xsmm::FusedBrgemmDispatchOp>(
+        loc, integer64, *dims,
+        xsmm::BinaryKindAttr::get(rewriter.getContext(), fusedMatch.binaryKind),
+        xsmm::UnaryKindAttr::get(rewriter.getContext(), fusedMatch.unaryKind),
+        getBrgemmFlags(rewriter, brgemmOp),
+        rewriter.getArrayAttr(xsmm::UnaryFlagsAttr::get(
+            rewriter.getContext(), xsmm::UnaryFlags::NONE)),
+        rewriter.getArrayAttr(
+            xsmm::BinaryFlagsAttr::get(rewriter.getContext(), *binaryFlags)),
+        dtype);
+
+    Value batchDim = rewriter.create<arith::ConstantOp>(
+        loc, integer64, rewriter.getIntegerAttr(integer64, batchSize));
+    SmallVector<Value, 6> invokeOperands;
+    invokeOperands.push_back(dispatched);
+    auto opItr = brgemmOp->getOperands().begin();
+    std::advance(opItr, 1);
```

Comment:
can you please add a comment here? Are you skipping the dispatch?

Reply:
Indeed, will add a comment